### PR TITLE
Revamp docs left nav

### PR DESCRIPTION
**Pull request by jhaaaa**
Preview: https://docs-xmtp-org-git-revamp-doc-nav-ephemerahq.vercel.app/

- Revamp left nav to create main sections based on the primary builder use cases
- Add a homepage that provides a tile per builder use case. The tile enables us to further highlight the use cases -- as well as provide a short description of each use case.
- Combine the existing 1:1 chat "Create a client" section with the existing group chat "Create a client" section to provide a common "Create a client" section that works for both 1:1 and group chat. 

### Create redirect
https://docs.xmtp.org/dms/client to https://docs.xmtp.org/client/create-client

To do:
- Reorg directories and files to match left nav structure. I will do this once we've agreed that the left nav structure is solid.
- I will continue to work on making V3 the default state of the docs.
- Known issue: When loaded, the left nav sections should be collapsed. This is a known issue with Vocs: https://github.com/wevm/vocs/issues/151

### CODE REVIEW

File: docs/pages/dms/client.mdx [hunk 1]
```diff
@@ -4,95 +4,147 @@ description: Learn how to create and configure an XMTP client
 
 # Create and configure an XMTP client
 
-The XMTP message API revolves around a network client that allows retrieving and sending messages to other network participants. A client must be connected to a wallet on startup. The client will request a wallet signature in two cases:
-
-1. To sign the newly generated key bundle. This happens only the very first time when key bundle is not found in storage.
-2. To sign a random salt used to encrypt the key bundle in storage. This happens every time the client is started (including the very first time).
+The XMTP message API revolves around a network client that allows retrieving and sending messages to other network participants. 
 
 ## Create a client
```
Comment (jhaaaa) [inline docs/pages/dms/client.mdx:9]: Updated this section to combine https://docs.xmtp.org/dms/client#create-a-client with https://docs.xmtp.org/groups/build-group-chat#create-a-client-that-supports-group-chat to provide a common "Create a client" section that works for both 1:1 and group chat.

Please review the code samples in particular - I asked Cursor for help with updating the samples.

File: docs/pages/dms/client.mdx [hunk 2]
```diff
@@ -750,7 +840,21 @@ let client = try await Client.create(account: account, options: clientOptions)
       <th>Parameter</th>
       <th>Default</th>
       <th>Description</th>
-      <th></th>
+    </tr>
```
Comment (jhaaaa) [inline docs/pages/dms/client.mdx:843]: Should the following options be added for RN?

        dbDirectory = "mydbdir", // Optional: specify database directory
        dbEncryptionKey = ByteArray(32) // Optional: 32-byte array for database

Comment (nplasterer) [reply docs/pages/dms/client.mdx:843]: dbEncryptionKey is not optional if you use V3

File: docs/pages/dms/client.mdx [hunk 3]
```diff
@@ -438,7 +490,23 @@ Set the `env` client option to `dev` while developing. Set it to `production` be
...
```
[the remark this answers is not in the document]
Comment (nplasterer) [reply docs/pages/dms/client.mdx:493]: cc @rygine

[the remark this answers is not in the document]
Comment (rygine) [reply docs/pages/dms/client.mdx:493]: yes, i think we should

File: docs/pages/dms/client.mdx [hunk 4]
```diff
@@ -4,95 +4,147 @@ description: Learn how to create and configure an XMTP client
...
+The XMTP message API revolves around a network client that allows retrieving and sending messages to other network participants. 
 
 ## Create a client
 
-A client is created that requires passing in a connected wallet that implements the Signer interface.
+To create an XMTP client, it must be connected to a wallet on startup. You must pass in a connected wallet that implements the Signer interface. This wallet is required for authentication and encryption purposes.
+
+For more information about wallet signature requests, see [Account signatures](/protocol/signatures).
+
+Clients created by XMTP v3 SDKs also support XMTP v2 messages. At a high-level, XMTP v3 is used to support group chat and XMTP v2 is used to support 1:1 chat.
```
Comment (nplasterer) [inline docs/pages/dms/client.mdx:15]: Can also support xmtp v2 messages if the wallet is an EOA.

File: docs/pages/dms/client.mdx [hunk 5]
```diff
@@ -4,95 +4,147 @@ description: Learn how to create and configure an XMTP client
...
+
+In the near future, XMTP v3 will support both group and 1:1 chat with Messaging Layer Security (MLS). Have questions or feedback, post to the [XMTP Community Forums](https://community.xmtp.org/c/development/58).
+
+Group chat is currently supported for mobile apps in Kotlin, Swift, React Native, and Node.js.
```
Comment (nplasterer) [inline docs/pages/dms/client.mdx:19]:
```suggestion
Group chat is currently supported in Kotlin, Swift, React Native, and Node.js.
```

File: docs/pages/dms/client.mdx [hunk 6]
```diff
@@ -4,95 +4,147 @@ description: Learn how to create and configure an XMTP client
...
+In the near future, XMTP v3 will also support group chat for the web.
+
+For more information about the work being done to support group chat for the web, see this [SQLite, WebAssembly, and Diesel](https://www.youtube.com/watch?v=qnCeXsqqKcY&t=240s) talk in the XMTP Developer Community Call.
+
+If your mobile app provides group chat, you must explicitly configure your client to enable this functionality by setting the `enableV3` client option to `true`.
+
+:::tip[Best practice: Set enableV3 to true]
 
-[Use client configuration options](#configure-the-client) to change parameters of a client's network connection.
+Regardless of whether your app provides group chat, set the `enableV3` client option to `true` to benefit from other aspects of the upgraded XMTP v3 protocol.
```
Comment (nplasterer) [inline docs/pages/dms/client.mdx:29]: Note you will not loose v2 functionality by doing this.

File: docs/pages/dms/client.mdx [hunk 7]
```diff
@@ -4,95 +4,147 @@ description: Learn how to create and configure an XMTP client
...
 :::code-group
 
 ```ts [JavaScript]
 import { Client } from "@xmtp/xmtp-js";
+
 // Create the client with a `Signer` from your application
-const xmtp = await Client.create(signer, { env: "dev" });
+const xmtp = await Client.create(signer, {
+  env: "production",
+  enableV3: true, // Enable group chat support
```
Comment (nplasterer) [inline docs/pages/dms/client.mdx:43]: Does not support v3 at the moment so probably shouldnt add here.

File: docs/pages/dms/client.mdx [hunk 8]
```diff
@@ -4,95 +4,147 @@ description: Learn how to create and configure an XMTP client
...
-    ClientOptions(
-        api = ClientOptions.Api(env = XMTPEnvironment.PRODUCTION, isSecure = true)
+val client = Client().create(
+    account = wallet, // wallet that implements SigningKey
+    options = ClientOptions(
+        api = ClientOptions.Api(XMTPEnvironment.PRODUCTION, isSecure = true),
+        enableV3 = true, // Enable group chat support
+        appContext = context, // Required if enableV3 is true
+        dbDirectory = "mydbdir", // Optional: specify database directory
+        dbEncryptionKey = ByteArray(32) // Optional: 32-byte array for database encryption
```
Comment (nplasterer) [inline docs/pages/dms/client.mdx:85]:
```suggestion
        dbEncryptionKey = ByteArray(32) // Required 32-byte array for database encryption if enableV3 is true
```

File: docs/pages/dms/client.mdx [hunk 9]
```diff
@@ -4,95 +4,147 @@ description: Learn how to create and configure an XMTP client
...
+// Inside your component
 export const CreateClient: React.FC<{ signer: Signer }> = ({ signer }) => {
   const { client, error, isLoading, initialize } = useClient();
 
   const handleConnect = useCallback(async () => {
     const options = {
       persistConversations: false,
-      env: "dev",
+      env: "production",
+      enableV3: true, // Enable group chat support
```
Comment (nplasterer) [inline docs/pages/dms/client.mdx:68]: Not supported in web so probably should be removed.

File: docs/pages/dms/client.mdx [hunk 10]
```diff
@@ -4,95 +4,147 @@ description: Learn how to create and configure an XMTP client
...
 
-// Create the client with a `SigningKey` from your app
 let client = try await Client.create(
-  account: account, options: .init(api: .init(env: .production)))
+    account: account, // account that implements SigningKey
+    options: .init(
+        api: .init(env: .production, isSecure: true),
+        enableV3: true, // Enable group chat support
+        dbDirectory: "mydir", // Optional: specify database directory
+        dbEncryptionKey: key // Optional: key for database encryption
```
Comment (nplasterer) [inline docs/pages/dms/client.mdx:99]:
```suggestion
        dbEncryptionKey: key // // Required 32-byte array for database encryption if enableV3 is true
```

File: docs/pages/dms/client.mdx [hunk 11]
```diff
@@ -4,95 +4,147 @@ description: Learn how to create and configure an XMTP client
...
 ```tsx [React Native]
 import { Client } from "@xmtp/xmtp-react-native";
-// Create the client with a `Signer` from your application
-const xmtp = await Client.create(signer);
+
+const xmtp = await Client.create(signer, {
+  env: "production",
+  enableV3: true, // Enable group chat support
+  dbDirectory: "mydbdir", // Optional: specify database directory
+  dbEncryptionKey: new Uint8Array(32), // Optional: 32-byte array for database encryption
```
Comment (nplasterer) [inline docs/pages/dms/client.mdx:111]:
```suggestion
  dbEncryptionKey: new Uint8Array(32), // Required 32-byte array for database encryption if enableV3 is true
```

File: docs/pages/dms/client.mdx [hunk 12]
```diff
@@ -612,104 +673,111 @@ Set the `env` client option to `dev` while developing. Set it to `production` be
...
+  </tr>
+  <tr>
+    <td>appContext</td>
+    <td>null</td>
+    <td>Required if `enableV3` is set to `true`.</td>
+  </tr>
+  <tr>
+    <td>dbEncryptionKey</td>
+    <td>null</td>
+    <td>Optional. If not provided, a key is created and stored for you.</td>
```
Comment (nplasterer) [inline docs/pages/dms/client.mdx:774]: This is now Required. We don't create or store keys anymore.

Comment (nplasterer) [reply docs/pages/dms/client.mdx:774]: We could link to the recommended key storage in kotlin https://developer.android.com/privacy-and-security/keystore

File: docs/pages/dms/client.mdx [hunk 13]
```diff
@@ -729,9 +797,31 @@ val client = Client().create(account = account, options = options)
...
+  </tr>
+  <tr>
+    <td>enableV3</td>
+    <td>`false`</td>
+    <td>To support group chat (a feature of XMTP v3), set to `true`. As a best practice, regardless of whether your app provides group chat, set `enableV3` to `true` to benefit from other aspects of the upgraded XMTP v3 protocol.</td>
+  </tr>
+    <tr>
+    <td>dbEncryptionKey</td>
+    <td>null</td>
+    <td>Optional. If not provided, the database remains unencrypted.</td>
```
Comment (nplasterer) [inline docs/pages/dms/client.mdx:818]: Required here as well. https://developer.apple.com/documentation/security/protecting-keys-with-the-secure-enclave

File: docs/pages/dms/client.mdx (hunk 2 above)
Comment (nplasterer) [reply docs/pages/dms/client.mdx:843]: But should be added to RN for sure.

File: docs/pages/dms/client.mdx (hunk 7 above)
Comment (jhaaaa) [reply docs/pages/dms/client.mdx:43]: @nplasterer Ah, thank you for this - I took it too far! I thought the guidance to have everyone enableV3 meant that at least the flag was available in all SDKs and everyone should set this value. But what we mean is that all apps built with Kotlin, Swift, RN, and Node should set enableV3 to true. GOT IT - thank you!

Comment (jhaaaa) [reply docs/pages/dms/client.mdx:43]: And the code comment is incorrect 😭. So sorry... I promise I know that group chat is not available for web yet.

File: docs/pages/dms/client.mdx (hunk 3 above)
[the remark this answers is not in the document]
Comment (jhaaaa) [reply docs/pages/dms/client.mdx:493]: Opened an issue for us to deliver this: https://github.com/xmtp/docs-xmtp-org/issues/37

I'll open a starter PR and share with you, @rygine  - thank you!

**Pull request merged**